### PR TITLE
Independent invalidation of user marks.

### PR DESCRIPTION
**Pull request by darina**

### CODE REVIEW

File: map/user_mark.hpp [hunk 1]
```diff
@@ -21,6 +21,8 @@ class UserMark : public df::UserPointMark
 {
   DISALLOW_COPY_AND_MOVE(UserMark);
 public:
+  uint16_t const kDefaultUserMarkProirity = 0xFFFF;
```
Comment (mpimenov) [inline map/user_mark.hpp]: constexpr (or define it in the cpp)

File: map/user_mark_container.hpp [hunk 2]
```diff
@@ -51,27 +51,29 @@ class UserMarkContainer : public df::UserMarksProvider
...
+  // If mark is found in "d" return distance from rect center.
+  // In multiple select choose mark with min(d).
   UserMark const * FindMarkInRect(m2::AnyRectD const & rect, double & d) const;
 
   static void InitStaticMarks(UserMarkContainer * container);
   static PoiMarkPoint * UserMarkForPoi();
   static MyPositionMarkPoint * UserMarkForMyPostion();
 
-  /// never save reference on UserMarksController
+  // Never save a reference on UserMarksController!
```
Comment (mpimenov) [inline map/user_mark_container.hpp]: s/on/to/ ?

**Comment by rokuz** [comment]
DUDE JTTP

**Comment by darina** [comment]
DUDE JTTP

**Pull request merged**